src/Components/Components/src/Services/IComponentContext.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.Services
+{
+    /// <summary>
+    /// Provides information about the environment in which components are executing.
+    /// </summary>
+    public interface IComponentContext
+    {
+        /// <summary>
+        /// Gets a flag to indicate whether there is an active connection to the user's display.
+        /// </summary>
+        /// <example>During prerendering, the value will always be false.</example>
+        /// <example>During server-side execution, the value can be true or false depending on whether there is an active SignalR connection.</example>
+        /// <example>During client-side execution, the value will always be true.</example>
+        bool IsConnected { get; }
```

Comment:
Should we document some guarantees about what happens when transition between states? Or is this not the right place for that.

Reply:
What kind of guarantees do you have in mind?

Reply:
Like, if you weren't connected when you last rendered - does that mean that you get called again when you are connected?

Reply:
Ah, I see. Since we're not proposing to do that, I'm not sure what there would be to say about it here.

I expect you know already, but the intent to deal with the mainstream "transition from prerendered to interactive" case is via lifecycle methods. For a more general "notify me when connection state changes", I'd expect that to be a separate API, which someone could wrap into a cascading value if they wanted.

Reply:
Ok cool. I guess what I was thinking about is where are we going to document to the pattern for using this property effectively. That's probably not a good fit for the in-code docs, so feel free to ignore.